src/append/file.rs

```diff
@@ -89,26 +90,51 @@ impl FileAppenderBuilder {
     /// will be resolved. Note that if the variable fails to resolve,
     /// $ENV{name_here} will NOT be replaced in the path.
     pub fn build<P: AsRef<Path>>(self, path: P) -> io::Result<FileAppender> {
-        let path_cow = path.as_ref().to_string_lossy();
-        let path: PathBuf = expand_env_vars(path_cow).as_ref().into();
-        if let Some(parent) = path.parent() {
+        let new_path = self.date_time_format(path);
+
+        if let Some(parent) = new_path.parent() {
             fs::create_dir_all(parent)?;
         }
         let file = OpenOptions::new()
             .write(true)
             .append(self.append)
             .truncate(!self.append)
             .create(true)
-            .open(&path)?;
+            .open(&new_path)?;
 
         Ok(FileAppender {
-            path,
+            path: new_path,
             file: Mutex::new(SimpleWriter(BufWriter::with_capacity(1024, file))),
             encoder: self
                 .encoder
                 .unwrap_or_else(|| Box::<PatternEncoder>::default()),
         })
     }
+
+    fn date_time_format<P: AsRef<Path>>(&self, path: P) -> PathBuf {
+        let path_cow = path.as_ref().to_string_lossy();
+        // Locate the start and end of the placeholder
+        if let Some(start) = path_cow.find('{') {
+            if let Some(end) = path_cow.find('}') {
+                // Extract the date format string
+                let date_format = &path_cow[start + 1..end];
```

Comment:
This won't work in the case where you have both an environment variable and the date unless the date come first. I.e. the example you updated in `Configuration.md` would fail this code because it would try to resolve the environment variable as your date format

Reply:
@bconn98 you are right. May I extract the env variable before extract date/time ?
The code will change to:

```
    fn date_time_format<P: AsRef<Path>>(&self, path: P) -> PathBuf {
        let path_cow = path.as_ref().to_string_lossy();
        // Extract the environment path first.
        let env_path: PathBuf = expand_env_vars(path_cow).as_ref().into();

        let date_time_path = env_path.to_str().unwrap();
        // Locate the start and end of the placeholder
        if let Some(start) = date_time_path.find('{') {
            if let Some(end) = date_time_path.find('}') {
                // Extract the date format string
                let date_format = &date_time_path[start + 1..end];
```

I tested with this change, this work with:
 - only ENV path
 - only date time format
 - combine ENV and date time format

Reply:
Yes this change looks better. A couple notes. 
1. You won't need to try and expand environment variables again at the end of the method.
2. What do you think about adding an identifier similar to how we did the ENV vars? That way it's easier visually to the user as well as easier to parse if we in the future want to add another pattern type to paths.

Reply:
1. Yes, I updated new commit to just return PathBuf instead of parsing the environment variable again.
2. That's great idea. we can easily handle our pattern in the future. But how do you think about:
 - We will release "date/time for log file" following the chrono format first for users (I see many developers need that feature including me)
 - Then, we will implement the new one that we will do the same as ENV vars to parse easier and can be handled by ourself